### PR TITLE
Update my profile endpoint

### LINKED ISSUE #12: Implement update my profile endpoint
**Description:**
This issue involves implementing an API to update the profile details. The endpoint should allow clients to make a PUT request to `{{baseUrl}}/api/me`.

sample response body:

```
{
    "created_at": "2023-06-29T09:15:00Z",
    "updated_at": "2023-06-30T16:45:00Z",
    "primary_email": "mentor@example.com",
    "contact_email": "mentor_contact@example.com",
    "first_name": "John",
    "last_name": "Doe",
    "image_url": "https://example.com/mentor_profile_image.jpg",
    "linkedin_url": "https://www.linkedin.com/in/johndoe",
    "type": "DEFAULT",
    "uuid": "12345678-1234-5678-1234-567812345678"
}
```
**Tasks:**
1. Create a controller for the `/controllers/profile` endpoint in the backend (create the route profile if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/profile` is implemented and accessible via a PUT request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by Shrenik0321**
## Purpose

The purpose of this PR is to fix #12 

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [ ] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/controllers/profile.controller.ts [hunk 1]
```diff
@@ -8,3 +10,69 @@ export const getProfile = async (req: Request, res: Response) => {
...
+  try {
+    const user: any = req.user;
+    const {
+      primary_email,
+      contact_email,
+      first_name,
+      last_name,
+      image_url,
+      linkedin_url,
+    }: any = req.body;
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:24]: The type should be Profile here,

```suggestion
    }: Profile = req.body;
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/controllers/profile.controller.ts [hunk 2]
```diff
@@ -8,3 +10,69 @@ export const getProfile = async (req: Request, res: Response) => {
     res.status(500).json({ error: err });
   }
 };
+
+export const updateProfile = async (req: Request, res: Response) => {
+  try {
+    const user: any = req.user;
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:16]: Let's avoid using any 

```suggestion
    const user: Profile = req.user;
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/controllers/profile.controller.ts (hunk 2 above)
Comment (Shrenik0321) [reply src/controllers/profile.controller.ts:16]: I'm unable to change the data type for this other than any..not sure why. Any solution?

File: src/services/profile.service.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        uuid: user.uuid,
+        primary_email: primary_email,
+      })
+      .getOne();
+
+    if (checkForExistingPrimaryEmail) {
+      return res
+        .status(409)
+        .json({ message: 'A similar primary_email already exists' });
+    }
```
Comment (anjula-sack) [inline src/services/profile.service.ts:40]: We don't need this check @Shrenik0321

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/services/profile.service.ts (hunk 3 above)
Comment (Shrenik0321) [reply src/services/profile.service.ts]: okay.Update made

File: src/routes/profile.route.test.ts [hunk 4]
```diff
@@ -41,4 +41,43 @@ describe('profile', () => {
...
+    beforeAll(async () => {
+      const testUser = {
+        email: 'test@gmail.com',
+        password: '123',
+      };
+
+      const response = await supertest(server)
+        .post('/api/auth/login')
+        .send(testUser)
+        .expect(200);
```
Comment (anjula-sack) [inline src/routes/profile.route.test.ts:57]: we need to create a user before login, then use this API. we will use db containers to run tests (not yet implemented due to the typeorm issue).

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Good job! @Shrenik0321

**Pull request merged**